artiq/devices/novatech409B/novatech409B.py

```diff
@@ -0,0 +1,362 @@
+                sys.exit()
+            self.__ser = serial.Serial(
+                serial_port_id,
+                baudrate=19200,
+                bytesize=8,
+                parity="N",
+                stopbits=1,
+                xonxoff=0,
+                timeout=0.05)
+        self.setup()
+
+    def __del__(self):
+        if not self.simulate_hw:
+            self.__ser.close()
+            time.sleep(1)
+
+    def echo(self, s):
+        ss = "novatech409B.echo() :: " + s
+        self.debug_message("echo", ss)
+        return ss
+
+    def debug_message(self, func_name, msg, level=2):
+        """generate debug message
+
+        :param str func_name: the calling function"s name
+        :param str msg: is a message
+        :param int level: is the debug level
+            ** 2 information helpful in typical use scenario
+            ** 3 full-on debug info (annoying)
+        :returns: None
+        """
+        if 1:  # (level <= self.__debug) :
+            # try a trick to automatically infer caller name
+            inferred_func_name = inspect.stack()[2][3]
+            s = inferred_func_name + "() :: " + msg
+            # here"s what a typical warnning message looks like:
+            # WARNING:ARTIQ.driver.novatech409B:
+            #      set_phase_continuous() :: M n
+            self.logger.warning(s)
+
+    def ser_send(self, myStr, ignore_unusual_response=False):
+        """send a string to the serial port
+
+        Routine for sending serial commands to device. It sends strings
+        and listens for a response terminated by a carriage return.
+
+        example:
+        ser_send("F0 1.0") #sets the freq of channel 0 to 1.0 MHz
+
+        :param str myStr: a character string to send to device
+        :returns: None
+        """
+        self.debug_message("ser_send", myStr, level=3)
+        s = myStr + "\r\n"
+        expected_response = b"OK\r\n"
+        result = b""
+        if self.simulate_hw is False:
+            try:
+                self.__ser.flush()
+                # after convert to python3 needed to cast between
+                # Python3x string and the expected bytes type
+                self.__ser.write(bytes(s, "UTF-8"))
+                time.sleep(self.serial_rw_delay)
+                result = self.__ser.read(1028)
+            except serial.SerialException as e:
+                self.debug_message("ser_send", e, level=0)
+
+            # check for error from device
+            # expected response (no error) is myStr\r\nOK\r\n
+            # after convert to python3 need to specify type of return
+            # to be bytes
+            if ~ignore_unusual_response:
+                if result != expected_response:
+                    print("ERROR :: novatech409B.ser_send() "
+                        "response was {}".format(result))
+                    return (result, expected_response)
+                    sys.exit()
```

Comment:
raise an exception if this is really needed. why would you want to exit the program here?